src/pdu.c

```diff
@@ -1047,24 +1047,28 @@ coap_pdu_parse_opt(coap_pdu_t *pdu) {
           if (!optsize || (COAP_PDU_IS_SIGNALING(pdu) ?
                !coap_pdu_parse_opt_csm(pdu, len) :
                !coap_pdu_parse_opt_base(pdu, len))) {
-            ok = ok && write_prefix(&obp, &outbuflen, "*", 1);
+            if (ok)
+              ok = write_prefix(&obp, &outbuflen, "*", 1);
             if (!optsize) {
               /* Skip to end of options to output all data */
               opt = pdu->token + pdu->used_size;
               length = 0;
             }
           }
           else {
-            ok = ok && write_prefix(&obp, &outbuflen, " ", 1);
+            if (ok)
+              ok = write_prefix(&obp, &outbuflen, " ", 1);
           }
           tlen = opt - opt_last;
           while (tlen--) {
-            ok = ok && write_char(&obp, &outbuflen, *opt_last, i);
+            if (ok)
+              ok = write_char(&obp, &outbuflen, *opt_last, i);
             opt_last++;
           }
         }
         if (length && *opt == COAP_PAYLOAD_START) {
-          ok = ok && write_char(&obp, &outbuflen, *opt, i);
```

Comment:
I do not see why Coverity barks on this line while not complaining about the previous uses of the same thing (possibly because ok is not used afterwards?). CID 1502424 is a false positive in my opinion.

Reply:
OK - reversing out the change.